test/Microsoft.AspNetCore.Hosting.Tests/WebHostBuilderTests.cs

```diff
@@ -573,6 +575,47 @@ public void Build_DoesNotOverrideILoggerFactorySetByConfigureServices()
             Assert.Equal(factory, factoryFromHost);
         }
 
+        [Fact]
+        public void StartupErrorsAreLoggedIfCaptureStartupErrorsIsTrue()
+        {
+            var testSink = new TestSink();
+            var factory = new TestLoggerFactory(testSink, enabled: true);
+            var builder = CreateWebHostBuilder()
+                .CaptureStartupErrors(true)
+                .ConfigureServices(services => services.AddSingleton<ILoggerFactory>(factory))
+                .Configure(app =>
+                {
+                    throw new InvalidOperationException("Startup exception");
+                })
+                .UseServer(new TestServer());
+
+            using (var host = (WebHost)builder.Build())
+            {
+                Assert.True(testSink.Writes.Any(w => w.Exception?.Message == "Startup exception"));
+            }
+        }
+
+        [Fact]
+        public void StartupErrorsAreLoggedIfCaptureStartupErrorsIsFalse()
+        {
+            var testSink = new TestSink();
+            var factory = new TestLoggerFactory(testSink, enabled: true);
+
+            var builder = CreateWebHostBuilder()
+                .CaptureStartupErrors(false)
+                .ConfigureServices(collection => collection.AddSingleton<ILoggerFactory>(factory))
+
+                .Configure(app =>
+                {
+                    throw new InvalidOperationException("Startup exception");
+                })
+                .UseServer(new TestServer());
+
+            Assert.Throws<InvalidOperationException>(() => builder.Build());
+
+            Assert.True(testSink.Writes.Any(w => w.Exception?.Message == "Startup exception"));
```

Comment:
`Assert.Contains("Startup exception", testSink.Writes.Select(w => w.Exception?.Message).Where(m => m != null));` is slightly better because it'll cause xunit to print the 2nd argument (the haystack). Helps when the test fails. With the current approach all you get is a `Assert.True(false)` which is less useful.